### PR TITLE
Adding sample security headers code for WP

### PR DESCRIPTION
**Pull request by carl-alberto**
Closes: #6158

## Summary

**[WordPress Best Practices](https://pantheon.io/docs/wordpress-best-practices#security-headers)** - Added a new section on security headers including plugins that can create them without `.htaccess`, and example code to set them yourself.

## Effect

Since Pantheon is Nginx based and the config cannot be modified to add security headers, it might be a good idea to have some sample code on how to do it in the platform since most of the results from search engines are solutions modifying .htaccess files and for Apache-based platforms

- Adding sample code for security headers in WP
- Some links to help users check their security headers

## Post Launch

- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | [Live](https://pantheon.io/docs) | [pr-6159](https://pr-6159--pantheon-docs.my.pantheonfrontend.website/docs) | Difference |
| --- | --- | --- | --- |
| Performance Score | `60` | `67` | `7` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `88` | `88` | `0` |

Lighthouse Audit Reports:

- [`pr-6159` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/25180/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/25180/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)


[Preview Environment](https://pr-6159--pantheon-docs.my.pantheonfrontend.website/docs)

**Review by alexfornuto — APPROVED** [review]
Thanks @carl-alberto! The sample code upgraded my site on securityheaders.com from an F to a C, but I'd still like to get a technical review from someone more fluent in security headers.

For now though, I've made some copy edits and would call the copy review complete.

**Comment by alexfornuto** [comment]
https://github.com/pantheon-systems/documentation/pull/6159/commits/fdfec99ed45d19cb0082b7fde3a4d831a4dd43fa Got my rating up to an A. Feature Policy has been replaced with Permissions Policy, and the `:` in the header name was being translated into the value for the header.

@carl-alberto how do you feel about the example values I provided for Permissions Policy?

**Comment by sarahg** [comment]
Good addition @carl-alberto!

A few things that might be worth adjusting:

- `no-referrer-when-downgrade` is the default value for `Referrer-Policy`, so setting it explicitly is not be necessary (this is the value you get if you don't specify anything at all). https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy
- Typo: `XX-XSS-Protection` should be `X-XSS-Protection`

It might be helpful to include a link to an external resource about HTTP headers in general

**Comment by alexfornuto** [comment]
After a long Slack discussion with @sarahg and @dficker, I've made a lot of changes. I'd like @carl-alberto to sign off on those changes, and I've written enough new content that it requires a new copy review. But tech review is covered.

File: source/content/wordpress-best-practices.md [hunk 1]
```diff
@@ -117,3 +117,29 @@ This method has the advantage of being toggleable without deploying code, by act
...
+Adding code like the example below in a plugin or to your theme can help add security headers for WordPress sites on Pantheon, or any other Nginx-based platform.
+
+These headers apply when accessing site pages or the REST API of WordPress, but not when directly accessing assets like `https://example.com/wp-content/uploads/2020/01/sample.json`.
+
+The code below is only an example to get you started. You'll need to modify it to match your needs, especially the Content Security Policy. Tools like [SecurityHeaders.com](https://securityheaders.com) can help to check your security headers and links to additional information on where to improve your security header profile.
+
+```php
+function additional_securityheaders( $headers ) {
+  if ( ! is_admin() ) {
+    $headers['Referrer-Policy']             = 'no-referrer-when-downgrade'; //This is the default value, the same as if it were not set.
```
Comment (carl-alberto) [inline source/content/wordpress-best-practices.md:134]: Maybe consistent spacing for the code sensitive user

**Review by carl-alberto — COMMENTED** [review]
Maybe a note

File: source/content/wordpress-best-practices.md (hunk 1 above)
Comment (alexfornuto) [reply source/content/wordpress-best-practices.md:134]: Addressed in https://github.com/pantheon-systems/documentation/pull/6159/commits/1c1beec0abb3bc2371a510f8b3d314d7f4432c0f

File: source/content/wordpress-best-practices.md [hunk 2]
```diff
@@ -117,3 +117,29 @@ This method has the advantage of being toggleable without deploying code, by act
   ```
 
 1. Commit your work, deploy code changes then activate the plugin on Test and Live environments.
+
+## Security Headers
+
+Pantheon's Nginx configuration cannot be modified to add security headers, and most solutions (including plugins) written about security headers for WordPress involve modifying the `.htaccess` file for Apache-based platforms.
+
+Adding code like the example below in a plugin or to your theme can help add security headers for WordPress sites on Pantheon, or any other Nginx-based platform.
```
[the remark this answers is not in the document]
Comment (alexfornuto) [reply source/content/wordpress-best-practices.md]: Addressed in https://github.com/pantheon-systems/documentation/pull/6159/commits/01d013d9e70d28f2262fecaf39fde3f27cb791ca - :+1: ?

[the remark this answers is not in the document]
Comment (carl-alberto) [reply source/content/wordpress-best-practices.md]: LGTM @alexfornuto !

File: source/content/wordpress-best-practices.md (hunk 1 above)
Comment (carl-alberto) [reply source/content/wordpress-best-practices.md:134]: 👍

**Review by carolyn-shannon — APPROVED** [review]
minor typo/caps fix, add a link. Looks great!

**Pull request merged**